lib/utils.pm

```diff
@@ -258,6 +259,21 @@ sub get_netboot_mirror {
     return get_var('MIRROR_' . uc($m_protocol));
 }
 
+sub zypper_call {
+    my $command = shift;
+    my $timeout = shift || 700;
+    my $str     = bmwqemu::hashed_string("ZN$command");
+
+    script_run("zypper -n $command; echo $str-\$?- > /dev/$serialdev", 0);
+
+    my $ret = wait_serial(qr/$str-\d+-/, $timeout);
+
+    if ($ret =~ /$str-(\d+)/) {
+        return $1;
+    }
```

Comment:
I'm not sure if enhanced original:
```perl
if ($ret && $ret =~ /$str-(\d+)/) {
    return $1;
}
```
is any more readable. I think my snippet is actually easier to understand. We expect our test writers to know regexes after all.

Explicit assignment may be the most understandable way:
```perl
if ($ret) {
  my ($ret_code) = $ret =~ /$str-(\d+)/;
  return $ret_code;
}
```